external-crates/move/move-compiler/src/cfgir/translate.rs

```diff
@@ -256,8 +257,152 @@ fn script(context: &mut Context, hscript: H::Script) -> G::Script {
 // Functions
 //**************************************************************************************************
 
+fn constants(
+    context: &mut Context,
+    module: Option<ModuleIdent>,
+    mut consts: UniqueMap<ConstantName, H::Constant>,
+) -> UniqueMap<ConstantName, G::Constant> {
+    let mut graph = DiGraphMap::new();
+    for (name, constant) in consts.clone() {
+        let deps = dependent_constants(&constant);
+        if deps.is_empty() {
+            graph.add_node(name);
+        } else {
+            for dep in deps {
+                graph.add_edge(name, dep, ());
+            }
+        }
+    }
+
+    let mut out_map = UniqueMap::new();
+    let mut sorted = petgraph_toposort(&graph, None);
+
+    while let Err(cycles) = sorted {
```

Comment:
I suppose this is fine for now, but maybe add a TODO for something faster :)

Reply:
I rewrote the algorithm to iterate over SCCs. Handling the neighbor cases is still a little bit of a mess because of how petgraph handles stuff, but it _should_ be more efficient and nicer now.